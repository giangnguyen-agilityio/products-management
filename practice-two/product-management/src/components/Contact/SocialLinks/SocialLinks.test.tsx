import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import SocialLink from '.'

describe('SocialLink component', () => {
  const props = {
    href: '#',
    icon: 'icon-url',
    alt: 'Social Icon',
  }

  it('renders link with correct attributes', () => {
    const { container, getByRole } = render(<SocialLink {...props} />)

    const linkElement = getByRole('link', { name: props.alt })

    expect(linkElement).toBeInTheDocument()
    expect(linkElement).toHaveAttribute('href', props.href)
    expect(container).toMatchSnapshot()
  })

  it('renders image with correct attributes', () => {
    const { getByAltText } = render(<SocialLink {...props} />)

    const imageElement = getByAltText(props.alt)

    expect(imageElement).toBeInTheDocument()
    expect(imageElement).toHaveAttribute('src', props.icon)
  })

  it('renders external link', () => {
    const { getByRole } = render(<SocialLink {...props} />)

    const linkElement = getByRole('link', { name: props.alt })

    expect(linkElement).toHaveAttribute('target', '_blank')
  })
})
