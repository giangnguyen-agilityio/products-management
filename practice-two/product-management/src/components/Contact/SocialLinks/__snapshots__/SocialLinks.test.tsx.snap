// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLink component renders link with correct attributes 1`] = `
<div>
  <a
    class="chakra-link css-eqjwk7"
    href="#"
    rel="noopener"
    target="_blank"
  >
    <img
      alt="Social Icon"
      class="chakra-image css-s6bwai"
      src="icon-url"
    />
  </a>
</div>
`;
